app/models/page.rb

```diff
@@ -9,4 +9,21 @@ class Page < ActiveRecord::Base
   validates :structure, presence: true
   validates :resource, presence: true
   validates :position, presence: true, uniqueness: { scope: :resource }
+
+  after_save :upsert_translation_elements
+
+  private
+
+  def upsert_translation_elements
+    Nokogiri::XML(structure).xpath('//content:text[@i18n-id]').each do |node|
+      onesky_phrase_id = node['i18n-id']
```

Comment:
No it's not; I realize I don't have any error checking there right now... and I'm fine with that for now